Gems/RobotecImGui/Code/Source/ImGui/SensorImGui.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Robotec.ai 2023. All rights reserved.
+ */
+
+#include "SensorImGui.h"
+
+#include <AzCore/Component/Entity.h>
+#include <ROS2/Sensor/Events/PhysicsBasedSource.h>
+#include <ROS2/Sensor/Events/TickBasedSource.h>
+#include <ROS2/Sensor/ROS2SensorComponentBase.h>
+#include <ROS2/Sensor/SensorConfiguration.h>
+
+#include <algorithm>
+
+namespace RobotecImGui
+{
+    AZStd::vector<ImGuiElement> SensorImGui::GetMessage()
+    {
+        m_imGuiElements.clear();
+        const auto childComponents = GetEntitySensorComponents(entity);
+        if (childComponents.empty())
+        {
+            return {};
+        }
+
+        for (const auto& component : childComponents)
+        {
+            std::shared_ptr<ROS2::ROS2SensorComponentBase<ROS2::TickBasedSource>> sharedSensorComponentTickBased;
+            std::shared_ptr<ROS2::ROS2SensorComponentBase<ROS2::PhysicsBasedSource>> sharedSensorComponentPhysicsBased;
+
+            if (auto* tickBased = azrtti_cast<ROS2::ROS2SensorComponentBase<ROS2::TickBasedSource>*>(component))
```

Comment:
Please move this code to another method for both implementations. At this level leave common logic.